StatefulSet IP preallocation cannot work as expected

**Describe the bug**
FlexibleIPAM e2e cases always failed
```
    --- FAIL: TestAntreaIPAM/testAntreaIPAMStatefulSetDedicated (18.02s)
    --- FAIL: TestAntreaIPAM/testAntreaIPAMStatefulSetShared (18.02s)
```
The StatefulSet Pods cannot use reserved IPs.
```
=== RUN   TestAntreaIPAM/testAntreaIPAMStatefulSetDedicated
    antreaipam_test.go:553: expectedIPAddressMap: {"192.168.240.130":{"ipAddress":"192.168.240.130","phase":"Allocated","owner":{"pod":{"name":"sts-test-atii00wt-0","namespace":"antrea-ipam-test","containerID":""},"statefulSet":{"name":"sts-test-atii00wt","namespace":"antrea-ipam-test","index":0}}},"192.168.240.131":{"ipAddress":"192.168.240.131","phase":"Allocated","owner":{"pod":{"name":"sts-test-atii00wt-1","namespace":"antrea-ipam-test","containerID":""},"statefulSet":{"name":"sts-test-atii00wt","namespace":"antrea-ipam-test","index":1}}}}
    antreaipam_test.go:584: IPPool status isn't correct: [{"ipAddress":"192.168.240.130","phase":"Reserved","owner":{"statefulSet":{"name":"sts-test-atii00wt","namespace":"antrea-ipam-test","index":0}}},{"ipAddress":"192.168.240.131","phase":"Reserved","owner":{"statefulSet":{"name":"sts-test-atii00wt","namespace":"antrea-ipam-test","index":1}}},{"ipAddress":"192.168.240.132","phase":"Allocated","owner":{"pod":{"name":"sts-test-atii00wt-0","namespace":"antrea-ipam-test","containerID":"28c3252057a5263e6da733d0818dea07e288fa971954ff3b49077c3f45d686d9"}}},{"ipAddress":"192.168.240.133","phase":"Allocated","owner":{"pod":{"name":"sts-test-atii00wt-1","namespace":"antrea-ipam-test","containerID":"601f5fdbc75710711e9330266a07aaca9c83b858cc3424c23d0bbcc5824352ef"}}}]
```

**To Reproduce**
Run FlexibleIPAM e2e with PR #6317

**Expected**
Pass all e2e cases.

**Actual behavior**
2 cases failed.

**Versions:**
main branch after PR #6317 merged

cc @luolanzone

--- a/pkg/util/k8s/transform.go
+++ b/pkg/util/k8s/transform.go
@@ -59,7 +59,6 @@ func TrimPod(obj interface{}) (interface{}, error) {
 		return obj, nil
 	}
 
-	pod.OwnerReferences = nil
 	pod.Spec.Volumes = nil
 	pod.Spec.InitContainers = nil
 	for i := range pod.Spec.Containers {

--- a/pkg/util/k8s/transform_test.go
+++ b/pkg/util/k8s/transform_test.go
@@ -86,6 +86,14 @@ func TestTrimK8sObject(t *testing.T) {
 					Name:      "test-pod",
 					Namespace: "default",
 					UID:       "test-uid",
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: "apps/v1",
+							Kind:       "DaemonSet",
+							Name:       "test-daemonset",
+							UID:        "5a39d3c8-0f5f-4aad-94bf-315c4fe11320",
+						},
+					},
 				},
 				Spec: corev1.PodSpec{
 					Containers: []corev1.Container{{

